src/components/composer/composer/abstract_composer_store.ts

```diff
@@ -609,6 +601,37 @@ export abstract class AbstractComposerStore extends SpreadsheetStore {
     return (token.parenthesesCode || "").startsWith(previousSymbolAtCursor.parenthesesCode || "");
   }
 
+  private computeParenthesisRelatedToCursor() {
+    if (
+      this.tokenAtCursor &&
+      ["LEFT_PAREN", "RIGHT_PAREN"].includes(this.tokenAtCursor.type) &&
+      this.tokenAtCursor.parenthesesCode !== ""
+    ) {
+      const matchingParenthesisIndexes = this.currentTokens.reduce((acc, t, index) => {
```

Comment:
I'm a bit tired so I didn't read all the function, but shouldn't this function be trivial ? If there is a parenthesis at the cursor, just loop on all the token to find a parenthesis with the same code and highlight them both.

I'm not sure how we could have multiple parenthesis with the same code ? Or your algorithm to add the code to the token is wrong ...

And we shoould mark the two parenthesis tokens and that's all, I don't think we should mark all the other tokens as not related to the cursor or w/e

Reply:
you must first ensure that the parentheses exist as a duo to highlight them. A parenthesis without its girlfriend means nothing. It therefore has no reason to be highlighted. It's the sad life of parentheses :'(

ok to not mark all, (my bad)

Reply:
Yeah but you know the parenthesis at the cursor, and you have its code. So you could just loop on all the tokens and if you find another parenthesis with the same code you highlight them both then return. I 'ond think we need to loop multuple times on the tokens ?

Reply:
in fact (in a simple version of the code) I will still need to go through the tokens twice, 

here we are in a case where the tokens are not necessarily reset: when you simply move the position cursor, the parentheses which were previously highlighted (isRelatedToCursorToken to true) must return to false 

in any case the whole code changes drastically in the following commit, it is only a passing state

I don't know if that really answers your question.